client/src/assets/sharedComponents/AlertDialog.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useCallback, useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+import Button from '@material-ui/core/Button';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+
+export default function AlertDialog(props: any) {
+  const { when, onConfirmation, onCancel } = props;
+
+  const history = useHistory();
+
+  const [showPrompt, setShowPrompt] = React.useState(false);
+  const [selectedRoute, setSelectedRoute] = React.useState('');
+
+  useEffect(() => {
+    if (when) {
+      history.block((prompt) => {
+        setSelectedRoute(prompt.pathname);
+        setShowPrompt(true);
+        return 'true';
```

Comment:
is this meant to be stringified? what's this return value doing?
remember to post any articles, documentation you used in your PR description plz

Reply:
It's intended, history.block will throw an error here if it's a boolean.

Reply:
updated description